### ISSUE TITLE
About calculating Logits and Reward

### ISSUE THREAD

**Issue by ChencongZJU** [dvlab-research/ARPO#11]
I saw this segment in the code:

<img width="859" height="58" alt="Image" src="https://github.com/user-attachments/assets/04e729bc-173f-44c2-83f9-46452947ea84" />

And by default, Verl sets inputs = prompts + response.

<img width="1260" height="567" alt="Image" src="https://github.com/user-attachments/assets/f2b84f6e-f538-4ff0-b917-63964eaebf62" />

I have two questions. First, in the current implementation for multi-turn dialogue, does the code extract the logits for each individual response turn? Also, why is batch['responses'] being assigned the value of batch['input_ids']?

**Comment by ChencongZJU** [comment]
Oh, I seem to understant that you use "response_mask" to get the right logits for action response instead of "response". And "reponse" just get logits for all tokens.

**Comment by FanbinLu** [comment]
Exactly. In this way we don't have to change the codes like compute_log_probs and many other functions.

**Comment by ChencongZJU** [comment]
Cool!
